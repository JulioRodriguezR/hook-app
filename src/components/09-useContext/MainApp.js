import React, { useState } from 'react'

import { UserContext } from './UserContext';
import { AppRouter } from './AppRouter';
import './styles.css';

export const MainApp = () => {

  const [user, setUser] = useState({})

  return (
    <UserContext.Provider value={{ user, setUser, temporal: '01' }}>
      <AppRouter />
    </UserContext.Provider>
  );

}
