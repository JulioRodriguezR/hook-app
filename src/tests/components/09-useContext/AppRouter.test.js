import React from 'react';
import { mount } from 'enzyme';

import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Test in <AppRouter />', () => {

  const user = {
    id: 1,
    name: 'USER' };

  const wrapper = mount(
    <UserContext.Provider value={{user}}>
      <AppRouter />
    </UserContext.Provider>
  );

  test('Should show correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

});