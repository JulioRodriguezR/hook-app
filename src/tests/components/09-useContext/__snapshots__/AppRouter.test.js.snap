// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <AppRouter /> Should show correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Navbar
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item active"
        >
          <a
            activeclassname="active"
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
          >
            Home 
            <span
              className="sr-only"
            >
              (current)
            </span>
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            activeclassname="active"
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            activeclassname="active"
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div>
    <h1>
      Home
    </h1>
    <hr />
    <pre>
      {
   "id": 1,
   "name": "USER"
}
    </pre>
  </div>
</div>
`;
